packages/etherscan/src/lib/etherscan.ts

```diff
@@ -130,6 +141,123 @@ async function getSourceCode(call: Etherscan, address: string) {
   });
 }
 
-////////////////
-//TRANSACTIONS//
-////////////////
+//////////////////
+// TRANSACTIONS //
+//////////////////
+
+function getStatus(call: Etherscan, txhash: string) {
+  return call<ExecutionStatusResult>({ module: 'transaction', action: 'getstatus', txhash });
+}
+
+function getTxReceiptStatus(call: Etherscan, txhash: string) {
+  return call<StatusResult>({ module: 'transaction', action: 'gettxreceiptstatus', txhash });
+}
+
+////////////
+// BLOCKS //
+////////////
+
+function getBlockReward(call: Etherscan, blockno: number) {
+  return call<BlockReward>({ module: 'block', action: 'getblockreward', blockno });
+}
+
+function getBlockCountdown(call: Etherscan, blockno: number) {
+  return call<BlockCountdown>({ module: 'block', action: 'getblockcountdown', blockno });
+}
+
+function getBlocknoByTime(call: Etherscan, timestamp: number, closest: Closest) {
+  return call<string>({ module: 'block', action: 'getblocknobytime', timestamp, closest });
+}
+
+//////////////////////////
+// BLOCKS: STATS MODULE //
+//////////////////////////
+
+// TODO : Verify if Date is the correct type
+function dailyAvgBlockSize(call: Etherscan, startdate: Date, enddate: Date, params: Optional<DailyAvgBlocksize>) {
+  return call<DailyAvgBlocksizeResponse[]>({
+    module: 'stats',
+    action: 'dailyavgblocksize',
+    startdate,
```

Comment:
You need to convert the date into a YYYY-MM-DD string format. Same for below.
You can create a utils function for that.